src/components/Button/Button.d.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { Omit } from '../../index';
+
+declare namespace Button {
+    export interface IButtonProps<T> extends Omit<React.HTMLProps<T>, 'size' | 'onClick'> {
+      className?: string;
+      size?: "sm" | "lg";
+      outline?: boolean;
+      link?: boolean;
+      block?: boolean;
+      children?: React.ReactNode;
+      disabled?: boolean;
+      color?: string;
+      square?: boolean;
+      pill?: boolean;
+      icon?: string;
+      social?: string;
+      loading?: boolean;
+      RootComponent?: React.ReactElement<any>;
```

Comment:
Using `React.ReactElement` will not allow strings to get passed to this prop. Prefer `React.ReactType` to match up to flow's `React.ElementType`